src/sentry/tasks/derive_code_mappings.py

```diff
@@ -0,0 +1,100 @@
+import logging
+from datetime import timedelta
+from typing import Any, List, Mapping, Optional, Set, Tuple
+
+from django.utils import timezone
+
+from sentry.db.models.fields.node import NodeData
+from sentry.models import Project
+from sentry.models.group import Group
+from sentry.models.organization import Organization, OrganizationStatus
+from sentry.tasks.base import instrumented_task
+from sentry.utils.safe import get_path
+
+ACTIVE_PROJECT_THRESHOLD = timedelta(days=7)
+GROUP_ANALYSIS_RANGE = timedelta(days=14)
+
+logger = logging.getLogger("sentry.tasks.derive_code_mappings")
+
+
+@instrumented_task(  # type: ignore
+    name="sentry.tasks.derive_code_mappings.identify_stacktrace_paths",
+    queue="derive_code_mappings",
+    max_retries=0,  # if we don't backfill it this time, we'll get it the next time
+)
+def identify_stacktrace_paths(
+    organizations: Optional[List[Organization]] = None,
+) -> Mapping[str, Mapping[str, List[str]]]:
+    """
+    Generate a map of projects to stacktrace paths for specified organizations,
+    or all active organizations if unspecified.
+
+    This filters out projects have not had an event in the last 7 days or have
+    non-python files in the stacktrace.
+    """
+    if organizations is None:
+        organizations = Organization.objects.filter(status=OrganizationStatus.ACTIVE)
+
+    filename_maps = {}
+    for org in organizations:
+        projects = Project.objects.filter(organization=org, first_event__isnull=False)
+        projects = [
+            project
+            for project in projects
+            if Group.objects.filter(
+                project=project,
+                last_seen__gte=timezone.now() - ACTIVE_PROJECT_THRESHOLD,
+            ).exists()
+        ]
+
+        project_file_map = {project.slug: get_all_stacktrace_paths(project) for project in projects}
+        filename_maps[org.slug] = project_file_map
+    return filename_maps
+
+
+def get_all_stacktrace_paths(project: Project) -> List[str]:
+    groups = Group.objects.filter(
+        project=project, last_seen__gte=timezone.now() - GROUP_ANALYSIS_RANGE
+    )
+
+    all_stacktrace_paths = set()
+    for group in groups:
+        event = group.get_latest_event()
```

Comment:
Doing this once per group will likely be quite slow since this will be making n+1 queries. I think you should be able to batch these by passing a list of group ids to a snuba query.

Reply:
Filed https://getsentry.atlassian.net/browse/WOR-2319 for this